include/zephyr/toolchain/common.h

```diff
@@ -106,6 +107,38 @@
 
 #endif /* _ASMLANGUAGE */
 
+/*
+ * General directive for assembly code, to align the following symbol, in bytes.
+ *
+ * Example:
+ *
+ *    BALIGN(4)
+ *    test_symbol:
+ *
+ * 'test_symbol' will get aligned to 4 bytes.
+ */
+
+#ifdef _ASMLANGUAGE
+
+  #if defined(CONFIG_X86) || defined(CONFIG_ARM) || defined(CONFIG_ARM64) || \
+	defined(CONFIG_NIOS2) || defined(CONFIG_RISCV) || \
+	defined(CONFIG_XTENSA) || defined(CONFIG_MIPS) || \
+	defined(CONFIG_ARCH_POSIX)
+
+    #define   BALIGN(x)   .balign   x
+
+  #elif defined(CONFIG_ARC) || defined(CONFIG_SPARC)
```

Comment:
For ARC classic there are multiple compilers, gnu and MetaWare. Metaware indeed doesn't support .balign, and .align works for both gnu and MetaWare. Should the condition for this choice be architecture, or rather toolchain variant? Maybe this shouldn't be in 'common.h' but specified per toolchain?